src/core/agents.jl

```diff
@@ -442,3 +444,33 @@ For an illustration of its usage see the [Tutorial](@ref).
 macro dispatch(f_def)
     return esc(:(DynamicSumTypes.@pattern $f_def))
 end
+
+"""
+    @finalize_dispatch
```

Comment:
This method is not referenced anywhere. In the `@dispatch` docstring you need to say: "if you are creating your own module/package that uses Agents.jl, and you are using `@dispatch` inside it, then you need to put 
```
[`@finalize_dispatch`](@ref)`()`
```
before the module end (but after all `@dispatch` calls).

Reply:
yes, this is necessary, did it